types/tabulator-tables/index.d.ts

```diff
@@ -7,6 +7,8 @@
 // tslint:disable:no-unnecessary-class
 // tslint:disable:no-empty-interface
 
+import type { WritingOptions } from 'xlsx';
```

Comment:
Please do not add another dependency, especially not if it is optional. The NPM dependency network is already a big mess, so we should not feed it until absolutely necessary.

Additionally, xlsx is not an allowed 3rd party package, according to the tests.

Reply:
This was only to maintain accurate types due to the new `writeOptions` option of the XLSX downloader using these options: https://tabulator.info/docs/5.5/release#download. A separate PR had already been submitted to DT-tools to add the package as allowed, but as it is not a native npm dependency it may not get added.

This leaves a few options and I'm not sure on the best method:

- Use the npm version, which is currently 2 minor versions behind, and if they're not longer publishing there, then it will only potentially get more behind.
- Add a new `@types/xlsx` package to DT, however the package provides its own types, so this would be largely redundant, other than to provide values to `tabulator-tables`, and would need maintaining separately and cause confusion.
- Copy over the types from their repo into the `index.d.ts` of `tabulator-tables`. There are 2 types that are actually used (`WritingOptions`, which extends `CommonOptions`) found [here](https://git.sheetjs.com/sheetjs/sheetjs/src/commit/485a4f30f17ee6e2d79680133251f0dbfff74c6d/types/index.d.ts#L256), however this would then mean that the `@types/tabulator-tables` package may need updating if/when `xlsx` is updated as well.
- Don't strongly type the `writeOptions` and just allow it to accept any object.

Reply:
I don't like adding the dependency either. Those options would change based off of the version your own version of sheet js, not tabulator. Tabulator doesn't have a hard dependency for it. I suggest leave it as any object.